src/elm/string.js

```diff
@@ -155,11 +154,9 @@ class Matches extends Expression {
     if (string == null || pattern == null) {
       return null;
     }
-    if (string.match(new RegExp(pattern))) {
-      return true;
-    } else {
-      return false;
-    }
+
+    const match = string.match(new RegExp(pattern));
+    return Boolean(match && match.length > 0 && match[0] === string);
```

Comment:
Your code works, but I think the code below would work too and is maybe a little more straightforward?
```suggestion
    return (new RegExp('^' + pattern + '$')).test(string);
```

Reply:
I'll test this out and verify it works, and if so then then I'll make the change!